### PR TITLE
Use a fully Rust based solution for rendering web page

### PR DESCRIPTION
**Pull request by ammaraskar**
Just a quick little draft I threw together as a solution for #114

This uses [askama](https://github.com/djc/askama) instead of Handlebars just because I think askama's integration with Rust is tighter and it lets you logically structure the templates (see the directory structure under src/web).

For now I have it set up to just spit out the fully rendered site into a `_site` folder and the code is a bit ugly right now but I'm not particularly attached to any of the design decisions here.

You can see a preview of the rendered site here: https://sharp-varahamihira-e1fb75.netlify.app/
It gets rid of the escaping issues in titles so now we see a nice little:

image

without having to take out the `<u8>` part.

One potential downside is that we do loose the automatic RSS feed that jekyll gave.

### CODE REVIEW

File: src/web/mod.rs [hunk 1]
```diff
@@ -0,0 +1,117 @@
+//! Code relating to the generation of the https://rustsec.org web site.
```
Comment (tarcieri) [inline src/web/mod.rs:1]: Small nit, but per [2018 edition path clarity](https://doc.rust-lang.org/edition-guide/rust-2018/module-system/path-clarity.html#no-more-modrs) this file should be called `web.rs`

**Comment by tarcieri** [comment]
Cool, looks like a great start!

I'm not terribly concerned about the RSS feed although it's a nice-to-have.

**Comment by tarcieri** [comment]
I'll see if I can address some of the nits and add back the RSS feed

**Comment by ammaraskar** [comment]
Aah thanks, I can also take a look when I get some time. Some of the other things I wanted to address were that it currently only runs from the project folder since it expects to copy the static assets from `src/web/static`.

**Comment by tarcieri** [comment]
@ammaraskar cool, if you'd like to open some PRs I probably won't look until this weekend

**Comment by ammaraskar** [comment]
> I'm not terribly concerned about the RSS feed although it's a nice-to-have.

After taking a second look at the RSS feed, it might actually be better to let it go. While we could use something like https://github.com/rust-syndication/rss to reimplement it, I don't think anyone actually uses it and it only contains the last 10 or so advisories anyway.

**Pull request merged**